src/trace.rs

```diff
@@ -469,6 +482,7 @@ impl Trace {
 
         let claims: &Vec<(AgentName, Claim)> = &ctx.claimer.deref().borrow().claims;
         if let Some(msg) = is_violation(claims) {
+            // [TODO] versus checking at each step ? Could detect violation earlier, before a blocking state is reached ? [BENCH] benchmark the efficiency loss of doing so
```

Comment:
Not sure if that can be done. But actually the idea way to check for violation at each step, because some violations maybe are only visible in-between a flight of messages.

Reply:
Maybe I did not understand how those checks are performed. It seemed to me that there were done once a trace is fully executed (?), in which case we might miss attacks, e.g., if an agent thinks that he has authenticated a peer, while he has not properly yet authenticated it.

Reply:
Oh sorry, yes, the checks are executed after the trace is done. But on the data which has been recorded during the trace execution. So no attacks should be missed.